src/components/MyNyplButton/MyNyplButton.js

```diff
@@ -104,13 +104,13 @@ class MyNyplButton extends React.Component {
    */
   renderMyNyplButton() {
     let buttonClass = '';
-    let iconClass = (HeaderStore.getMyNyplVisible()) ? 'nypl-icon-solo-x' : 'nypl-icon-wedge-down';
+    let iconClass = (this.state.visible) ? 'nypl-icon-solo-x' : 'nypl-icon-wedge-down';
     const icon = (<span className={`${iconClass} icon`} style={styles.MyNyplIcon}></span>);
     const labelColorClass = (this.props.isLoggedIn) ? ' loggedIn' : '';
     const myNyplButtonLabel = (this.props.patronName) ? 'My Account' : 'Log In';
     const loggedInFadeInAnimation = (this.props.patronName) ? ' animated fadeIn' : '';
 
-    if (HeaderStore.getMyNyplVisible()) {
+    if (this.state.visible) {
```

Comment:
doesn't iconClass already have (this.state.visible) check in line 107? Maybe we can consolidate a single if-else condition assignment for button and icon classes. As well as `this.props.patronName` which is occurring twice.

Reply:
Updated.